crates/biome_js_analyze/src/lint/nursery/no_enum.rs

```diff
@@ -0,0 +1,51 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::TsEnumDeclaration;
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Disallow TypeScript enum.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// enum Foo {
+    ///     BAR = 'bar'
+    /// }
+    /// ```
+    ///
```

Comment:
You can add a valid case like below?

```
    /// ### Valid
    ///
    /// ```js
    /// const Foo = {
    ///     BAR: 'bar',
    /// };
    /// ```
```